build.gradle

```diff
@@ -2,91 +2,99 @@ import se.bjurr.gitchangelog.plugin.gradle.GitChangelogTask
 
 buildscript {
 	repositories {
-		maven { url = "https://files.minecraftforge.net/maven" }
+		maven { url = 'https://files.minecraftforge.net/maven' }
 		jcenter()
 		mavenCentral()
 	}
 	// This is only here while FG is under active development
 	configurations.classpath {
 		resolutionStrategy {
-			cacheDynamicVersionsFor 10, "seconds"
-			cacheChangingModulesFor 0, "seconds"
+			cacheDynamicVersionsFor 10, 'seconds'
+			cacheChangingModulesFor 0, 'seconds'
 		}
 	}
 	dependencies {
-		classpath group: "net.minecraftforge.gradle", name: "ForgeGradle", version: "3.+", changing: true
+		classpath group: 'net.minecraftforge.gradle', name: 'ForgeGradle', version: '4.1.+', changing: true
 	}
 }
 
 plugins {
-	id "com.matthewprenger.cursegradle" version "1.4.0"
-	id "se.bjurr.gitchangelog.git-changelog-gradle-plugin" version "1.64"
+	id 'java'
+	id 'eclipse'
+	id 'idea'
+	id 'com.matthewprenger.cursegradle' version '1.4.0'
+	id 'se.bjurr.gitchangelog.git-changelog-gradle-plugin' version '1.65'
 }
-apply from: "buildtools/ColoredOutput.gradle"
-apply plugin: 'idea'
-apply plugin: "net.minecraftforge.gradle"
+apply plugin: 'net.minecraftforge.gradle'
+apply from: 'buildtools/ColoredOutput.gradle'
 
 //adds the build number to the end of the version string if on a build server
 String build_number = System.getenv().BUILD_NUMBER
 if (build_number == null) {
-	build_number = "9999"
+	build_number = '9999'
 }
 
 String semver = "${version_major}.${version_minor}.${version_patch}"
 version = "${semver}.${build_number}"
-group = "mezz.jei" // http://maven.apache.org/guides/mini/guide-naming-conventions.html
-
-sourceCompatibility = targetCompatibility = JavaVersion.VERSION_1_8
-compileJava {
-	sourceCompatibility = targetCompatibility = JavaVersion.VERSION_1_8
-}
-
-repositories {
-	maven { url 'https://repo1.maven.org/maven2' }
-	maven { url 'https://maven.latmod.com/' }
-}
+group = 'mezz.jei' // http://maven.apache.org/guides/mini/guide-naming-conventions.html
 
 sourceSets {
 	api
-	main
-	test
+	main {
+		compileClasspath += sourceSets.api.output
+		runtimeClasspath += sourceSets.api.output
+		resources {
+			srcDir 'src/generated/resources'
+		}
+	}
+	test {
+		compileClasspath += sourceSets.api.output
+		runtimeClasspath += sourceSets.api.output
+	}
 }
 
 configurations {
-	apiCompile.extendsFrom(compile)
+	apiImplementation.extendsFrom(implementation)
+	apiRuntimeOnly.extendsFrom(runtimeOnly)
+}
+
+sourceCompatibility = targetCompatibility = compileJava.sourceCompatibility = compileJava.targetCompatibility = JavaVersion.VERSION_1_8
```

Comment:
For FG4 given it uses Gradle 6.8.+ this should be `java.toolchain.languageVersion = JavaLanguageVersion.of(8)`

Reply:
The toolchains are annoying because they override your set JDK in the IDE and download new JDKs into your (already huge) .gradle folder.
I don't need to download Java 8 if I have it installed already, thanks for nothing Gradle :)